kairon/actions/definitions/pyscript.py

```diff
@@ -0,0 +1,102 @@
+from typing import Text, Dict, Any
+
+from loguru import logger
+from mongoengine import DoesNotExist
+from rasa_sdk import Tracker
+from rasa_sdk.executor import CollectingDispatcher
+
+from kairon.actions.definitions.base import ActionsBase
+from kairon.shared.actions.data_objects import ActionServerLogs, PyscriptActionConfig
+from kairon.shared.actions.exception import ActionFailure
+from kairon.shared.actions.models import ActionType
+from kairon.shared.actions.utils import ActionUtility
+from kairon.shared.constants import KaironSystemSlots
+
+
+class ActionPyscript(ActionsBase):
+
+    def __init__(self, bot: Text, name: Text):
+        """
+        Initialize Pyscript action.
+
+        @param bot: bot id
+        @param name: action name
+        """
+        self.bot = bot
+        self.name = name
+        self.__response = None
+        self.__is_success = False
+
+    def retrieve_config(self):
+        """
+        Fetch Pyscript action configuration parameters from the database
+
+        :return: PyscriptActionConfig containing configuration for the action as a dict.
+        """
+        try:
+            pyscript_action_config_dict = PyscriptActionConfig.objects().get(bot=self.bot, name=self.name,
+                                                                             status=True).to_mongo().to_dict()
+            logger.debug("pyscript_action_config: " + str(pyscript_action_config_dict))
+            return pyscript_action_config_dict
+        except DoesNotExist as e:
+            logger.exception(e)
+            raise ActionFailure("No pyscript action found for given action and bot")
+
+    async def execute(self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict[Text, Any]):
+        """
+        Retrieves action config and executes it.
+        Information regarding the execution is logged in ActionServerLogs.
+
+        @param dispatcher: Client to send messages back to the user.
+        @param tracker: Tracker object to retrieve slots, events, messages and other contextual information.
+        @param domain: Bot domain
+        :return: Dict containing slot name as keys and their values.
+        """
+        pyscript_action_config = None
+        bot_response = None
+        exception = None
+        status = "SUCCESS"
+        dispatch_bot_response = False
+        filled_slots = {}
+        msg_logger = []
+
+        try:
+            pyscript_action_config = self.retrieve_config()
+            tracker_data = ActionUtility.build_context(tracker, True)
+            dispatch_bot_response = pyscript_action_config['dispatch_response']
+            source_code = pyscript_action_config['source_code']
+            bot_response = ActionUtility.run_pyscript(source_code, tracker_data)
```

Comment:
as asked in the last review, we need to pass the complete context as predefined variables and just return the slots and bot response back. we also decided on a format